src/Backups/BackupCoordinationFileInfos.cpp

```diff
@@ -88,19 +106,66 @@ void BackupCoordinationFileInfos::prepare() const
             info.data_file_index = i;
             info.base_size = 0; /// Base backup must not be used while creating a plain backup.
             info.base_checksum = 0;
-            total_size_of_files += info.size;
+
+            if (info.reference_target.empty())
+            {
+                file_name_to_info.emplace(info.file_name, &info);
+                total_size_of_files += info.size;
```

Comment:
We can use this `copy-inside-backup` approach for other copies too if we insert here something like:

```
SizeAndChecksum size_and_checksum{info.size, info.checksum};
auto [it, inserted] = data_file_index_by_checksum.emplace(size_and_checksum, i);
if (inserted)
{
    info.data_file_name = info.file_name;
    info.data_file_index = i;
}
else
{
    /// specify that info must not be written in the usual way ...
    info.data_file_name.clear();
    info.data_file_index = -1;

    /// ... because target_info will copied to its place instead
    auto & target_info = it->second;
    target_info->reference_sources.push_back(info.file_name);
}
```

Reply:
so we are okay with using size and checksum in plain backups as unique id?

Reply:
A plain backup means every file is stored in its place: there can files of the same contents; empty files are stored too; and references to the base backup are not used. So a plain backup is about how it must look on Disk or S3 when it's completed. It's not how we should write it - we can write it any way we like if the result is the same. I mean there is no difference in the result whether we
```
copy SRC TO DEST[0], DEST[1], ... DEST[N]
```
or
```
copy SRC TO DEST[0], then
copy DEST[0] to DEST[1], ... DEST[N]
```

Reply:
but there is a difference if we use size and checksum
if files from different backup entries match in size and checksum this will produce a different result

Reply:
> if files from different backup entries match in size and checksum this will produce a different result
I didn't quite understand. Do you mean that two files of the same size but different by contents can have the same checksum?

Reply:
yes

Reply:
I mean chances are almost nonexistent for a collision to happen, especially with 128bits we use for checksum but just asking if we are okay with that.

Reply:
The formula is
```
p(n, H) = 1-exp(-n^2 / 2 / H)
```
So if we used 128 hashes without size, for a backup with 1000000 files we would have the probability like `10^-27` that two hashes match. With sizes the probability must be even smaller.

Reply:
Though that optimization is not very crucial anyway.

Reply:
let me do this in a different PR so we are not blocking backups of KeeperMap and then we can discuss if we are okay with the solution.